bullet_train-super_scaffolding/lib/scaffolding/routes_file_manipulator.rb

```diff
@@ -54,17 +54,23 @@ def divergent_parts
   def find_namespaces(namespaces, within = nil)
     namespaces = namespaces.dup
     results = {}
-    block_end = Scaffolding::BlockManipulator.find_block_end(starting_from: within, lines: lines) if within
-    lines.each_with_index do |line, line_number|
-      if within
-        next unless line_number > within
-        return results if line_number >= block_end
-      end
-      if line.include?("namespace :#{namespaces.first} do")
-        results[namespaces.shift] = line_number
-      end
-      return results unless namespaces.any?
+    reinstantiate_masamune_object
+
+    # `within` can refer to either a `resources`, `namespace`, `scope`, or `shallow` block.
+    blocks = @msmn.method_calls.select { |node| node.token_value.match?(/resources|namespace|scope|shallow/) }
+    namespace_nodes = blocks.select { |node| node.token_value.match?(/namespace/) }
+
+    if within
+      starting_block = blocks.find { |block| block.line_number - 1 == within }
+      block_range = (starting_block.location.start_line - 1)..(starting_block.location.end_line - 1)
```

Comment:
I actually don't think we'll have to do `- 1` here because the block range is checking against `node.line_number`

Reply:
Dang off-by-one errors! 🤘